feature(#155): 마이페이지 마크업 및 UserCollection 컴포넌트 구현

suubiin opened the pull request:
### PR 유형

- [x] 새로운 기능 추가
- [x] 파일 혹은 폴더명 수정

### PR 체크 리스트

<!-- PR이 다음 요구 사항을 충족하는지 확인하세요. -->

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트)

### 🔎 작업 내용

<!-- 어떤 작업을 하셨는지 상세하게 작성해주세요. -->

- 마이페이지 마크업
- UserCollection 컴포넌트 구현
- 컴포넌트 명 변경 UserStatus -> UserActivity
  <br />

### 이슈

<!-- 이슈 키워드와 함께 #을 입력한 후 이슈 번호를 선택해주세요. -->
<!-- 예시 : resolves #1 -->

resolves #155

suubiin committed:
feature: 마이페이지 1차 마크업

suubiin committed:
Merge branch 'develop' into feature/mypage-choi

suubiin committed:
Merge branch 'develop' into feature/mypage-choi

suubiin committed:
마이페이지 중간 커밋

suubiin committed:
Merge branch 'develop' into feature/mypage-choi

suubiin committed:
UserCollection 컴포넌트 중간 커밋

suubiin committed:
Merge branch 'develop' into feature/mypage-choi

suubiin committed:
Merge branch 'develop' into feature/mypage-choi

suubiin committed:
feature: 마이페이지 마크업 및 UserCollection 컴포넌트 구현
- 마이페이지 마크업
- UserCollection 컴포넌트 구현
- 컴포넌트 명 변경 UserStatus -> UserActivity

suubiin committed:
Merge branch 'develop' into feature/mypage-choi

woolen-h commented on src/components/User/UserCollection.tsx:
```diff
@@ -0,0 +1,46 @@
+import { Link } from 'react-router-dom';
+
+interface UserCollectionItemPops {
```
어라...? Pops?

woolen-h commented on src/components/User/UserCollection.tsx:
```diff
@@ -0,0 +1,46 @@
+   text: string;
+}
+
+function UserCollectionItem({ to, iconClass, text }: UserCollectionItemPops) {
```
```tsx
const UserCollectionItem = memo(
   ({ to, iconClass, text }: UserCollectionItemProps) => (
   ```
   로 저장하여 프롭스가 변동 없을 경우 리렌더링 이슈 방지하면 어떨까 싶습니다!

woolen-h commented on src/components/User/UserCollection.tsx:
```diff
@@ -0,0 +1,46 @@
+         iconClass: 'fi-rr-comment-alt text-green-600',
+         text: '나의 리뷰',
+      },
+   ];
```
```tsx
const items = useMemo(
      () => [
         {
            to: '/my-page/favorites',
            iconClass: 'fi-rr-heart text-red-600',
            text: '나의 찜',
         },
         {
            to: '/my-page/reviews',
            iconClass: 'fi-rr-comment-alt text-green-600',
            text: '나의 리뷰',
         },
      ],
      []
   );
   ```
   만약 위에서 memo를 사용하는 방식으로 가게 되면 `useMemo`를 사용하는 쪽으로 변경해주시면 될 것 같습니다!

woolen-h commented on src/components/Main/Layout.tsx:
```diff
@@ -31,4 +31,4 @@ export default function Layout({ hasBackBtn, hasLogo, title }: LayoutProps) {
       </>
    );
-}
+}
```
마지막 행 추가 부탁드립니다 ! 깃헙 상에서 오류로 뜨는 표시입니다!

woolen-h commented on src/pages/Settings/MyPage.tsx:
```diff
@@ -1,5 +1,52 @@
+         <div className="mb-3 flex flex-col items-center">
+            <img
+               src=""
+               alt="프로필 이미지"
```
`<img alt=` 속성은 스크린 리더에서 이미지라는 기본 설명을 붙이기 때문에 `alt` 내 값에서는 제외해도 될 것 같습니당!

woolen-h commented on src/pages/Settings/MyPage.tsx:
```diff
@@ -1,5 +1,52 @@
+                  title="평균"
+                  count={4.5}
+                  style={{ gridColumn: 'span 2' }}
+               />
```
```tsx
const activities = [
   { title: '찜 개수', count: 999 },
   { title: '리뷰 개수', count: 999 },
   { title: '평균', count: 4.5, style: { gridColumn: 'span 2' } }
];

<div className="grid grid-cols-2 gap-4">
   {activities.map((activity, index) => (
      <UserActivity key={index} {...activity} />
   ))}
</div>
```
반복되는 부분은 객체 & map으로 뿌려주면 어떨까 싶습니당! 확인 부탁드립니다!

happyhye reviewed (approved):
검토 완료 👍
한울님 코멘트 반영해주시면 될 것 같아요~~

suubiin committed:
Merge branch 'develop' into feature/mypage-choi

suubiin committed:
modify: 피드백 반영 수정
- Layout: 프리티어 적용
- UserActivity: 시맨틱 마크업으로 수정
- UserCollection: 오타 수정/ UserCollectionItem에 memo 적용/ 변수 함수 밖으로 이동
- MyPage: 이미지 alt 속성 '프로필'로 수정/ 링크 버튼 컴포넌트 handleClick 속성 제거/ 반복되는 UserActivity 컴포넌트 map 함수 사용 방식으로 수정 및 변수 객체로 수정 후 함수 밖에서 관리

woolen-h reviewed (approved):
고생하셨습니다~
